suffuse/examples_test.go

```diff
@@ -0,0 +1,88 @@
+package suffuse
+
+import (
+  "github.com/satori/go.uuid"
+  "os"
+  "os/exec"
+  "time"
+)
+
+func ExampleSed() {
+  with(func (in string, out string) {
+    execute := executeIn(out)
+
+    // Fill a file with digits 1 to 10.
+    pipeToFile(command("seq", "1", "10"), in + "/seq.txt")
+
+    // Mount a directory through suffuse.
+    startFuse("suffuse", "-m", out, in)
+
+    //List the contents through the suffuse mount.
+    execute("ls")
+
+    // It's a 10 line file, one number to a line.
+    execute("wc", "-l", "seq.txt")
+
+    // Via suffuse, a derived file ending with #4,6p is a sed
+    // command executed on the actual file.
+    execute("cat", "seq.txt#4,6p")
+
+    // Arbitrary sed commands, different sized files.
+    execute("ls", "-g", "-G", "seq.txt#5,10p")
+
+    // These files are effectively indistinguishable from "real" files.
+    execute("ls", "-g", "-G", "seq.txt#1,3p")
+
+    //Output:
+    // seq.txt
+    // 10 seq.txt
+    // 4
+    // 5
+    // 6
+    // -rw-rw-r-- 1 13 Jul 10  2015 seq.txt#5,10p
+    // -rw-rw-r-- 1 6 Jul 10  2015 seq.txt#1,3p
+
+  })
+}
```

Comment:
This looks awesome given what's possible inside go's syntactic limitations, but given that this is really nothing but shelling out, does it offer some benefit over something along the lines of 
```go
executeBash(`
  ls                      # List the contents through the suffuse mount.
  wc -l seq.txt           # It's a 10 line file, one number to a line.
  cat 'seq.txt#4,6p'      # Via suffuse, a derived file ending with #4,6p is...
  ls -gG 'seq.txt#5,10p'  # Arbitrary sed commands, different sized files.
  ls -gG 'seq.txt#1,3p'   # These files are effectively indistinguishable from "real" files.
`)
```
AFAIK go ships with no string interpolation. I'm sure there are a bunch of named string interpolation libraries which plus-or-minus call "ReplaceAll" and call it a day. Anyway, with or without interpolation I feel like the direct expression of these commands is superior (even if you want to keep all the whitespace.)